lib/sidekiq/client.rb

```diff
@@ -248,10 +248,10 @@ def raw_push(payloads)
     def atomic_push(conn, payloads)
       if payloads.first.key?("at")
         conn.zadd("schedule", payloads.flat_map { |hash|
-          at = hash.delete("at").to_s
+          at = hash["at"].to_s
           # ActiveJob sets this but the job has not been enqueued yet
           hash.delete("enqueued_at")
-          [at, Sidekiq.dump_json(hash)]
+          [at, Sidekiq.dump_json(hash.except("at"))]
```

Comment:
AFAIK this creates a copy of the original hash, which obviously has implications. Not sure if there are better ways to handle this obscure edge case without impacting the happy path.

Reply:
Good analysis, I don't see an easy solution without significant refactoring which breaks compatibility. The Client API feels pretty hacked together these days so any change is difficult. We'll have to accept the additional cost for now.